File: challenges_java/src/main/java/ch01_intro/example1.java
package ch01_intro;

public class example1 {

    public static String stringReversal(String input) {
        return input.length() <= 1 ? input : stringReversal(input.substring(1)) + input.charAt(0);
    }

    public static void main(String[] args) {

        long startTime = System.nanoTime();
        var result = stringReversal("Lenchen");
        long elapsedTime = System.nanoTime() - startTime;
        System.out.printf("ExecutionTime: %d ms / Result: %s%n", elapsedTime/1000000, result);


    }
}

File: challenges_python/src/ch01_intro/example1.py
def string_reversal(input_string: str) -> str:
    if not isinstance(input_string, str):
        raise TypeError("Sorry, wrong input-type. Please input string.")

    return input_string[::-1]


if __name__ == '__main__':
    import time

    value = 'Lenchen'

    t = time.process_time()
    result = string_reversal(value)
    # exception = string_reversal(123)
    elapsed_time = time.process_time() - t

    print(f"ExecutionTime: {elapsed_time} ms / Result: {result}")
    # print(f"ExecutionTime: {elapsed_time} ms / Result: {exception}")
